### PR TITLE
feat(cluster): Allow to deploy multiple poolers

### PR DESCRIPTION
**Pull request by dragoangel**
implements: https://github.com/cloudnative-pg/charts/issues/360

### CODE REVIEW

**Comment by itay-grudev** [comment]
Can you sync with me on the CloudNativePG slack?

I can't review these until the weekend, but I can give you guidance, so it's as easy as possible.

**Comment by dragoangel** [comment]
Hi @itay-grudev, sure 😊

**Comment by dragoangel** [comment]
In short about this PR: as it contains **breaking change** it will be merged a bit later.

File: charts/cluster/templates/pooler.yaml [hunk 1]
```diff
@@ -1,21 +1,25 @@
-{{ if .Values.pooler.enabled }}
+{{ if gt (len .Values.poolers) 0 }}
```
Comment (itay-grudev) [inline charts/cluster/templates/pooler.yaml:1]: Why do we need the outside check? The `range` should be sufficient.

File: charts/cluster/templates/pooler.yaml [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
+{{ $context := . -}}
```
Comment (itay-grudev) [inline charts/cluster/templates/pooler.yaml:2]: No need to cache the context. Inside the `range` you can access the top context by using `$`. Example: `$.Values`

File: charts/cluster/test/monitoring/01-monitoring_cluster.yaml [hunk 3]
```diff
@@ -18,8 +18,9 @@ cluster:
...
-  enabled: true
-  instances: 1
-  monitoring:
-    enabled: true
+poolers:
+  - name: rw
+    type: rw
+    instances: 1
+    monitoring:
+      enabled: true
```
Comment (itay-grudev) [inline charts/cluster/test/monitoring/01-monitoring_cluster.yaml:26]: Add at least two poolers to the test to verify the code above generates the YAML correctly.

**Review by itay-grudev — CHANGES_REQUESTED** [review]

File: charts/cluster/test/monitoring/01-monitoring_cluster.yaml (hunk 3 above)
Comment (dragoangel) [reply charts/cluster/test/monitoring/01-monitoring_cluster.yaml:26]: I added 2 poolers into poolers test, was expect it was enough, but sure, can add here too.

Comment (itay-grudev) [reply charts/cluster/test/monitoring/01-monitoring_cluster.yaml:26]: Oh! I didn't realize that. But this is also good. Thanks.

**Review by itay-grudev — APPROVED** [review]

File: charts/cluster/templates/pooler.yaml (hunk 2 above)
Comment (dragoangel) [reply charts/cluster/templates/pooler.yaml]: To note: I taken it from scheduled backups, I think we should change it there as well?

**Comment by itay-grudev** [comment]
@dragoangel You're missing the monitoring section in the per-pooler configuration, hence the error. This wasn't a problem before since all the properties were always set. The best way to fix that would be an extensive use of `default` I suppose.

P.S. Because the old property is `pooler` and the new one `poolers`, this can be made backwards compatible.

**Comment by dragoangel** [comment]
@itay-grudev you mean in auto tests?

**Comment by dragoangel** [comment]
Hi @itay-grudev I pushed proper label example and tests.

I see 3 options:
1. leave as is
2. move `.Values.cluster.monitoring` and `.Values.pooler.monitoring` to `.Values.monitoring`, and leave podMonitor relablings under `cluster` and each `pooler` if they need to get this settings.
3. remove `.Values.poolers.[0].monitoring.enabled` and `.Values.poolers.[0].monitoring.podMonitor.enabled` and take this info from `cluster`, while still looks for `relablings` under specific pooler. It will assume - if we create pod monitor for cluster we want to have pod monitor for pooler too, but this not always can be true?

For me all 3 options are okay, but 1 and 2 looks better.

**Review by itay-grudev — APPROVED** [review]

**Comment by itay-grudev** [comment]
@dragoangel FYI: I went with option 1 as it was the simplest.

**Pull request merged**